app/Http/Controllers/PostController.php

```diff
@@ -0,0 +1,112 @@
+<?php
+
+namespace App\Http\Controllers;
+
+
+use Illuminate\Http\Request;
+use App\Http\Requests\PostRequest;
+use App\Repository\Eloquent\PostRepository;
+
+class PostController extends Controller
+{
+    protected $postRepository;
+
+    public function __construct(PostRepository $postRepository)
+    {
+        $this->postRepository = $postRepository;
+    }
+
+    public function home(){
+        $paginatedPost = $this->postRepository->getPaginatedPost();
```

Comment:
there is no need for this temporary variable, as it doesn't help explain anything, the below makes sense by itself

`return view("home", ["posts" => $this->postRepository->getPaginatedPost()]);`